commit a5e3aaa654c15760afdfb85d0b6fd825ce068efc
Author: Appana Durga Kedareswara rao <appana.durga.rao@xilinx.com>
Date:   Mon Jun 27 12:10:22 2022 +0530

    microblaze: Add xmb_manager_register function
    
    Triple Modular Redundancy (TMR) Microblaze solution provides soft error
    injection, detection, correction and recovery for Microblaze cores in the
    system. The Xilinx/AMD Triple Modular Redundancy (TMR) solution in Vivado
    provides all the necessary building blocks to implement a redundant
    triplicated MicroBlaze subsystem. This processing subsystem is
    fault-tolerant and continues to operate nominally after encountering an
    error. Together with the capability to detect and recover from errors,
    the implementation ensures the reliability of the entire subsystem.
    
    When the break vector gets asserted because of error injection,
    the break signal must be blocked before exiting from the break handler,
    This commit adds support for xmb_manager_register api which updates the
    TMR manager address and control register and error count and reset callback
    function arguments, which will be used by the break handler to block the
    break and call the error count callback function and reset callback
    function.
    
    Signed-off-by: Appana Durga Kedareswara rao <appana.durga.rao@xilinx.com>
    Link: https://lore.kernel.org/r/20220627064024.771037-2-appana.durga.rao@xilinx.com
    Signed-off-by: Michal Simek <michal.simek@amd.com>

diff --git a/arch/microblaze/Kconfig b/arch/microblaze/Kconfig
index 996132a5ef35..4ebb56d6d959 100644
--- a/arch/microblaze/Kconfig
+++ b/arch/microblaze/Kconfig
@@ -204,6 +204,16 @@ config TASK_SIZE
 	hex "Size of user task space" if TASK_SIZE_BOOL
 	default "0x80000000"
 
+config MB_MANAGER
+	bool "Support for Microblaze Manager"
+	depends on ADVANCED_OPTIONS
+	help
+	  This option enables API for configuring the MicroBlaze manager
+	  control register, which is consumed by the break handler to
+	  block the break.
+
+	  Say N here unless you know what you are doing.
+
 endmenu
 
 menu "Bus Options"